skactiveml/classifier/_wrapper.py

```diff
@@ -322,3 +350,874 @@ def __getattr__(self, item):
+        self.metric_dict = metric_dict
+        self.estimator = estimator
+        self.class_prior = class_prior
+        self.frequency_max_fit_len = frequency_max_fit_len
+        self.frequency_replacing_strategy = frequency_replacing_strategy
+
+    def fit(self, X, y, sample_weight=None, **fit_kwargs):
+        """Fit the model using X as training data and y as class labels.
+
+        Parameters
+        ----------
+        X : matrix-like, shape (n_samples, n_features)
+            The sample matrix X is the feature matrix representing the samples.
+        y : array-like, shape (n_samples) or (n_samples, n_outputs)
+            It contains the class labels of the training samples.
+            Missing labels are represented the attribute 'missing_label'.
+            In case of multiple labels per sample (i.e., n_outputs > 1), the
+            samples are duplicated.
+        sample_weight : array-like, shape (n_samples) or (n_samples, n_outputs)
+            It contains the weights of the training samples' class labels. It
+            must have the same shape as y.
+        fit_kwargs : dict-like
+            Further parameters as input to the 'fit' method of the 'estimator'.
+
+        Returns
+        -------
+        self: KernelFrequencyClassifier,
+            The KernelFrequencyClassifier is fitted on the training data.
+        """
+        self._fit_frequency_estimator(
+            fit_function="fit",
+            X=X,
+            y=y,
+            sample_weight=sample_weight,
+            **fit_kwargs,
+        )
+        return self._fit(
+            fit_function="fit",
+            X=X,
+            y=y,
+            sample_weight=sample_weight,
+            **fit_kwargs,
+        )
+
+    @if_delegate_has_method(delegate="estimator")
+    def partial_fit(self, X, y, sample_weight=None, **fit_kwargs):
+        """Partially fitting the model using X as training data and y as class
+        labels.
+
+        Parameters
+        ----------
+        X : matrix-like, shape (n_samples, n_features)
+            The sample matrix X is the feature matrix representing the samples.
+        y : array-like, shape (n_samples) or (n_samples, n_outputs)
+            It contains the class labels of the training samples.
+            Missing labels are represented the attribute 'missing_label'.
+            In case of multiple labels per sample (i.e., n_outputs > 1), the
+            samples are duplicated.
+        sample_weight : array-like, shape (n_samples) or (n_samples, n_outputs)
+            It contains the weights of the training samples' class labels. It
+            must have the same shape as y.
+        fit_kwargs : dict-like
+            Further parameters as input to the 'fit' method of the 'estimator'.
+
+        Returns
+        -------
+        self : KernelFrequencyClassifier,
+            The KernelFrequencyClassifier is fitted on the training data.
+        """
+        self._fit_frequency_estimator(
```

Comment:
high potential for redundancies. maybe start with estimator.fit() and then access ExtLabelEnc and classes from estimator for determining the frequencies. Thereby, we do not need to check if classes and labelencoder are similar